Library/ViewModels/PledgePaymentMethodsViewModelTests.swift

```diff
@@ -106,7 +121,83 @@ final class PledgePaymentMethodsViewModelTests: TestCase {
         [true, true, true, true, true, true, false, true],
         [true, true, true, true, true, true, true, false, true]
       ])
-      self.reloadPaymentMethodsProjectCountry.assertValues(["Brooklyn, NY", "Brooklyn, NY"])
+      self.reloadPaymentMethodsProjectCountry.assertValues([
+        (0...response.me.storedCards.nodes.count - 1).map { _ in "Brooklyn, NY" },
+        (0...response.me.storedCards.nodes.count).map { _ in "Brooklyn, NY" }
+      ], "New and available card added")
+      self.reloadPaymentMethodsSelectedCard.assertValues([
+        response.me.storedCards.nodes.first,
+        userCreditCard
+      ], "Newly added card is selected")
+
+      self.vm.inputs.creditCardSelected(paymentSourceId: userCreditCard.id)
+
+      self.updateSelectedCreditCard.assertValues([userCreditCard])
+    }
+  }
+
+  func testCardFromBackingIsSelectedFirst() {
+    let response = UserEnvelope<GraphUserCreditCard>(me: GraphUserCreditCard.template)
+    let mockService = MockService(fetchGraphCreditCardsResponse: response)
+    let userCreditCard = GraphUserCreditCard.visa
+
+    let project = Project.cosmicSurgery
+      |> Project.lens.state .~ .live
+      |> Project.lens.personalization.isBacking .~ true
+      |> Project.lens.personalization.backing .~ (
+        .template
+          |> Backing.lens.paymentSource .~ Backing.PaymentSource.visa
+          |> Backing.lens.status .~ .pledged
+          |> Backing.lens.reward .~ Reward.postcards
+          |> Backing.lens.rewardId .~ Reward.postcards.id
+          |> Backing.lens.shippingAmount .~ 10
+          |> Backing.lens.amount .~ 700
+      )
+
+    withEnvironment(apiService: mockService, currentUser: User.template) {
+      self.reloadPaymentMethodsCards.assertDidNotEmitValue()
+      self.reloadPaymentMethodsAvailableCardTypes.assertDidNotEmitValue()
+      self.reloadPaymentMethodsProjectCountry.assertDidNotEmitValue()
+      self.reloadPaymentMethodsSelectedCard.assertDidNotEmitValue()
+
+      self.vm.inputs.configureWith((User.template, project, false))
+      self.vm.inputs.viewDidLoad()
+
+      self.scheduler.run()
+
+      let visaFirstCards = response.me.storedCards.nodes.sorted {
+        card1, _ in card1 == GraphUserCreditCard.visa
+      }
+
+      self.reloadPaymentMethodsCards.assertValues([visaFirstCards])
+      self.reloadPaymentMethodsAvailableCardTypes.assertValues([
+        [true, true, true, true, true, true, false, true]
+      ])
+      self.reloadPaymentMethodsProjectCountry.assertValues([
+        (0...response.me.storedCards.nodes.count - 1).map { _ in "Hastings, UK" }
+      ], "One card is unavailable")
```

Comment:
Is this message meant to go in the assertion above on line 173?

Reply:
I added this mostly to indicate why I'm doing the range as `0...response.me.storedCards.nodes.count - 1`, the existing tests didn't have any messages describing the assertions really. I could add the same message to all of the `reloadPaymentMethodsAvailableCardTypes` but I think it's pretty clear what's going on?

Reply:
Ah gotcha!